src/models/pressure-system.ts

```diff
@@ -87,14 +94,18 @@ export class PressureSystem {
     center.lat = Math.max(minLat, center.lat);
     const step = config.minPressureSystemDistance / 100;
     const minDistRes = minDistToOtherSystems(center, otherPressureSystems);
-    const heading = minDistRes.heading;
+    const heading = minDistRes.heading || 0; // Provide default heading if null
     let minDist = minDistRes.minDist;
     // Why is an iterative approach used instead of a single calculation based on the minPressureSystemDistance
     // and heading? Note this single calculation could result in new center being to close to another pressure system.
     // Iterative calculations ensure that we'll always end up away from *all* the pressure systems. Initial heading
     // is used, so the interaction feels a bit more natural for user.
     while (minDist < config.minPressureSystemDistance) {
-      center = moveTo(center, { distance: step, heading: heading + 180 });
+      const newPos = moveTo(
+        {lat: center.lat, lon: center.lng},
+        { distance: step, heading: heading + 180 }
+      );
+      center = { lat: newPos.lat, lng: newPos.lon };
```

Comment:
There's some inconsistency with spaces, but mostly they seem to be skipped, so:
```suggestion
        {lat: center.lat, lon: center.lng},
        {distance: step, heading: heading + 180}
      );
      center = {lat: newPos.lat, lng: newPos.lon};
```